components/nimbus/src/stateful/evaluator.rs

```diff
@@ -47,6 +51,10 @@ impl From<AppContext> for TargetingAttributes {
 }
 
 impl TargetingAttributes {
+    pub(crate) fn with_recorded_context(&mut self, recorded_context: Arc<dyn RecordedContext>) {
```

Comment:
Since you're not actually doing anything with the `Arc` part of the pointer, you don't need to take a `Arc<dyn RecordedContext>`. You can take a `&dyn RecordedContext` instead (which means you dont need to use `Arc` in your tests). `Arc<T>` implements `Deref` so you can get a `&T` via `&*arc` (or call `.as_ref()`)